test/e2e/accounts/common.ts

```diff
@@ -0,0 +1,340 @@
+import { privateToAddress } from 'ethereumjs-util';
+import FixtureBuilder from '../fixture-builder';
+import {
+  PRIVATE_KEY,
+  PRIVATE_KEY_TWO,
+  WINDOW_TITLES,
+  clickSignOnSignatureConfirmation,
+  convertETHToHexGwei,
+  switchToOrOpenDapp,
+  unlockWallet,
+  validateContractDetails,
+} from '../helpers';
+import { Driver } from '../webdriver/driver';
+
+export const TEST_SNAPS_SIMPLE_KEYRING_WEBSITE_URL =
+  'https://metamask.github.io/snap-simple-keyring/1.0.1/';
+
+export const ganacheOptions = {
+  accounts: [
+    {
+      secretKey: PRIVATE_KEY,
+      balance: convertETHToHexGwei(25),
+    },
+    {
+      secretKey: PRIVATE_KEY_TWO,
+      balance: convertETHToHexGwei(25),
+    },
+  ],
+};
+
+export const accountSnapFixtures = (title: string | undefined) => {
+  return {
+    dapp: true,
+    fixtures: new FixtureBuilder()
+      .withPermissionControllerConnectedToTestDapp(false)
+      .withPreferencesController({
+        disabledRpcMethodPreferences: {
+          eth_sign: true,
+        },
+      })
+      .build(),
+    ganacheOptions,
+    failOnConsoleError: false,
+    title,
+  };
+};
+
+// convert PRIVATE_KEY to public key
+export const PUBLIC_KEY = privateToAddress(
+  Buffer.from(PRIVATE_KEY.slice(2), 'hex'),
+).toString('hex');
+
+export async function installSnapSimpleKeyring(
+  driver: Driver,
+  isAsyncFlow: boolean,
+) {
+  await driver.navigate();
+
+  await unlockWallet(driver);
+
+  // navigate to test Snaps page and connect
+  await driver.openNewPage(TEST_SNAPS_SIMPLE_KEYRING_WEBSITE_URL);
+  await driver.clickElement('#connectButton');
+
+  await driver.delay(500);
+
+  await driver.switchToWindowWithTitle(WINDOW_TITLES.Notification);
+
+  await driver.delay(500);
+
+  await driver.clickElement({
+    text: 'Connect',
+    tag: 'button',
+  });
+
+  await driver.clickElementSafe('[data-testid="snap-install-scroll"]');
+
+  await driver.waitForSelector({ text: 'Install' });
+
+  await driver.clickElement({
+    text: 'Install',
+    tag: 'button',
+  });
+
+  await driver.waitForSelector({ text: 'OK' });
+
+  await driver.clickElement({
+    text: 'OK',
+    tag: 'button',
+  });
+
+  await driver.switchToWindowWithTitle(WINDOW_TITLES.SnapSimpleKeyringDapp);
+
+  await driver.waitForSelector({
+    text: 'Connected',
+    tag: 'span',
+  });
+
+  if (isAsyncFlow) {
+    await toggleAsyncFlow(driver);
+  }
+}
+
+async function toggleAsyncFlow(driver: Driver) {
+  await driver.switchToWindowWithTitle(WINDOW_TITLES.SnapSimpleKeyringDapp);
+
+  // click the parent of #use-sync-flow-toggle (trying to click the element itself gives "ElementNotInteractableError: could not be scrolled into view")
+  await driver.clickElement({
+    xpath: '//input[@id="use-sync-flow-toggle"]/..',
```

Comment:
Is it possible to use testId here? Using xpath can make this part brittle if there are minor changes in the test snap layout.

Reply:
Unfortunately there's no `testid` to grab here.  And if I try
```
await driver.clickElement('input#use-sync-flow-toggle');
```
the result is
```
ElementNotInteractableError: Element <input id="use-sync-flow-toggle" class="Toggle__ToggleInput-sc-ag66wo-1 emMiBJ" type="checkbox"> could not be scrolled into view
```
We need the parent of that `input`, and it doesn't have a `data-testid`.